postgresql/postgresql_client_test.go

```diff
@@ -0,0 +1,181 @@
+package postgresql
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/internal/config"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/internal/util"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/resources"
+	pg "github.com/Azure/azure-sdk-for-go/services/preview/postgresql/mgmt/2020-02-14-preview/postgresqlflexibleservers"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/marstr/randname"
+)
+
+var (
+	serverName = generateName("gosdkpostgresql")
+	dbName     = "postgresqldb1"
+	dbLogin    = "postgresqldbuser1"
+	dbPassword = generatePassword("postgresqldbuserpass!1")
+)
+
+func addLocalEnvAndParse() error {
+	// parse env at top-level (also controls dotenv load)
+	err := config.ParseEnvironment()
+	if err != nil {
+		return fmt.Errorf("failed to add top-level env: %+v", err)
+	}
+	return nil
+}
+
+func addLocalFlagsAndParse() error {
+	// add top-level flags
+	err := config.AddFlags()
+	if err != nil {
+		return fmt.Errorf("failed to add top-level flags: %+v", err)
+	}
+
+	flag.StringVar(&serverName, "pgsqlServerName", serverName, "Name for PostgreSQL server.")
+	flag.StringVar(&dbName, "pgsqlDbName", dbName, "Name for PostgreSQL database.")
+	flag.StringVar(&dbLogin, "pgsqlDbUsername", dbLogin, "Username for PostgreSQL login.")
+	flag.StringVar(&dbPassword, "pgsqlDbPassword", dbPassword, "Password for PostgreSQL login.")
+
+	// parse all flags
+	flag.Parse()
+	return nil
+}
+
+func setup() error {
+	var err error
+	err = addLocalEnvAndParse()
+	if err != nil {
+		return err
+	}
+	err = addLocalFlagsAndParse()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func teardown() error {
+	if config.KeepResources() == false {
+		// does not wait
+		_, err := resources.DeleteGroup(context.Background(), config.GroupName())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// test helpers
+func generateName(prefix string) string {
+	return strings.ToLower(randname.GenerateWithPrefix(prefix, 5))
+}
+
+// Just add 5 random digits at the end of the prefix password.
+func generateName(pass string) string {

```

Comment:
I believe this should be a `generatePassword`? since we already have a `generateName` above.
```suggestion
func generatePassword(pass string) string {

```

Reply:
Fixed